jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/AttributeNormalizer.java

```diff
@@ -411,17 +418,29 @@ private String expand(String prefix, String property, String suffix)
         {
             if (property.equals(attr.key))
             {
-                try
+                if (attr.resource != null)
                 {
-                    String uri = prefix + attr.value + suffix;
-                    Resource resource = ResourceFactory.root().newResource(uri);
-                    if (resource.exists())
-                        return uri;
+                    Resource resource = attr.resource.resolve(suffix);
+                    if (resource != null && resource.exists())
+                        return prefix + attr.value + suffix;
                 }
-                catch (Exception ex)
+                else
                 {
-                    if (LOG.isDebugEnabled())
-                        LOG.trace("ignored", ex);
+                    try
+                    {
+                        String uri = prefix + attr.value + suffix;
+                        try (ResourceFactory.Closeable resourceFactory = ResourceFactory.closeable())
```

Comment:
@lorban I think we need to avoid needing this section of code. It's a major waste of time to convert to a `Resource` just to do exists when the caller only wants a string, and is probably then going to do their own conversions on this.  If we never allow the `Resource` inside a `URIAttribute` to be null, we can just delete this code.

Reply:
How about if there is only 1 resource, we just expand it.  Then we only go to this code if there are more than 1 and no resources.